spark/src/main/scala/org/apache/iceberg/spark/SparkTableUtil.scala

```diff
@@ -259,7 +326,7 @@ object SparkTableUtil {
       partitionPath: Map[String, String],
       partitionUri: String,
       conf: Configuration,
-      metricsSpec: MetricsConfig = MetricsConfig.getDefault): Seq[SparkDataFile] = {
+      metricsSpec: MetricsConfig): Seq[SparkDataFile] = {
```

Comment:
Why remove this?

Reply:
It is a private method and I think it is better to explicitly require a metrics config to be sure the caller really wants to pass `MetricsConfig.getDefault`.